sdk/servicebus/service-bus/src/receivers/receiverCommon.ts

```diff
@@ -186,7 +218,14 @@ function settleMessage(
 
   let error: Error | undefined;
   if (message.delivery.remote_settled) {
-    error = new Error(MessageAlreadySettled);
+    const condition = isDefined(message.sessionId)
+      ? ErrorNameConditionMapper.SessionLockLostError
+      : ErrorNameConditionMapper.MessageLockLostError;
```

Comment:
I like that this will now be a MessagingError...just curious how you came up with these choices for the condition. They seem reasonable but was there prior art?

Reply:
When a message is settled, the lock is given up as well. So, this is the closest "amqp" error we have to throw when someone tries to settle a message that is already settled. Deferred messages don't have the luxury of having `delivery`, so this check will only apply to non deferred messages.

@richardpark-msft What would happen if we skip this check altogether and rely on the service throwing the relevant error? Also worth checking what other languages are doing here. I vividly recall discussing these scenarios and having a big table with all cases

Reply:
I am slightly hesitant in throwing SessionLockLostError here though... 
When a non session message is settled, its lock is given up.
When a session message is settled, the lock on the session is not given up.

So, if the user had a logic somewhere to catch SessionLockLostError and recreate the session, they will be doing so unnecessarily. Moreover, that attempt may fail as the session lock is already held by another receiver

Reply:
I'll revert it to the previous error without condition (since we have disagreement about it) but I don't want to hold this particular PR on it since it's not changing existing behavior (we were already throwing a non-retryable error)

Reply:
Thanks! Just saw the commit, so will resolve this thread